invenio_oauthclient/handlers/rest.py

```diff
@@ -0,0 +1,396 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Handlers for customizing oauthclient endpoints."""
+
+from __future__ import absolute_import, print_function
+
+from functools import wraps
+
+from flask import abort, current_app, jsonify, make_response, \
+    redirect, render_template, request, session, url_for
+from flask_login import current_user
+from invenio_db import db
+from six.moves.urllib.parse import parse_qs, urlencode, urlsplit, urlunsplit
+
+from ..errors import AlreadyLinkedError, OAuthCERNRejectedAccountError, \
+    OAuthClientError, OAuthError, OAuthRejectedRequestError
+from ..models import RemoteAccount
+from ..proxies import current_oauthclient
+from ..signals import account_info_received, account_setup_committed, \
+    account_setup_received
+from ..utils import oauth_authenticate, oauth_get_user, rest_oauth_register
+from .utils import get_session_next_url, response_token_setter, token_getter, \
+    token_session_key, token_setter
+
+
+def response_handler_postmessage(remote, url, payload=None):
+    """Postmessage response handler."""
+    return render_template(
+        'invenio_oauthclient/postmessage.html',
+        payload=payload
+    )
+
+
+def default_response_handler(remote, url, payload=None):
+    """Default response handler."""
+    scheme, netloc, path, query, fragment = urlsplit(url)
+    scheme = scheme or request.scheme
+    netloc = netloc or request.host
+    assert netloc
```

Comment:
```suggestion
```

Since we don't require an absolute URL (in `invenio-accounts` we wanted to generate a URL to use in an email sen to the user), it's fine if there is no `netloc` or `scheme`.